libfreerdp/core/nego.c

```diff
@@ -1440,3 +1440,12 @@ void nego_free_nla(rdpNego* nego)
 	nla_free(nego->transport->nla);
 	nego->transport->nla = NULL;
 }
+
+const BYTE* nego_get_routing_token(rdpNego* nego, DWORD* RoutingTokenLength)
+{
+	if (!nego)
```

Comment:
Just an idea but perhaps it would be nice to also set `*RoutingTokenLength` to 0 ?